### ISSUE TITLE
LESS not output to <style> in production build

### ISSUE THREAD

**Issue by sgrove** [gatsbyjs/gatsby#2422]
Starting from `gatsby-starter-blog`, we're a bit blocked by Gatsby not outputting CSS in a production build (just an empty gatsby `<style id="gatsby-inlined-css"></style>` tag). The output `public/styles.css` is also completely empty.

We're using the [less](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-less/src/gatsby-node.js) plugin, which works perfectly well in development mode (the `<style id="gatsby-inlined-css">` tag has all of the appropriate CSS). The plugin itself is fairly small, but I'm not familiar with the underlying abstractions, so having a hard time figuring out why it's not working for us. What should we be looking at in the plugin/build pipeline to figure out the difference between dev/prod builds here?

`gatsby-config.js`
```javascript
module.exports = {
  siteMetadata: {
    title: 'P',
    author: 'P',
  },
  plugins: [
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        path: `${__dirname}/src/pages`,
        name: 'pages',
      },
    },
    {
      resolve: `gatsby-transformer-remark`,
      options: {
        plugins: [
          {
            resolve: `gatsby-remark-images`,
            options: {
              maxWidth: 590,
            },
          },
          {
            resolve: `gatsby-remark-responsive-iframe`,
            options: {
              wrapperStyle: `margin-bottom: 1.0725rem`,
            },
          },
          'gatsby-remark-prismjs',
          'gatsby-remark-copy-linked-files',
        ],
      },
    },
    {
      resolve: `gatsby-plugin-google-analytics`,
      options: {
        //trackingId: `ADD YOUR TRACKING ID HERE`,
      },
    },
    `gatsby-plugin-react-helmet`,
    {
      resolve: `gatsby-plugin-less`,
      options: {
        stage: `develop`,
        options: {autoprefix: '> 5%'},
      },
    },
  ],
};
```

`package.json`
```javascript
{
  "name": "gatsby-starter-blog",
  "description": "Starter Gatsby Blog",
  "version": "1.0.0",
  "author": "Kyle Mathews <mathews.kyle@gmail.com>",
  "bugs": {
    "url": "https://github.com/gatsbyjs/gatsby-starter-blog/issues"
  },
  "dependencies": {
    "classnames": "^2.2.5",
    "gatsby": "^1.9.42",
    "gatsby-link": "^1.6.8",
    "gatsby-plugin-google-analytics": "^1.0.4",
    "gatsby-plugin-less": "^1.0.5",
    "gatsby-plugin-manifest": "^1.0.8",
    "gatsby-plugin-preact": "^1.0.3",
    "gatsby-plugin-react-helmet": "^1.0.3",
    "gatsby-remark-copy-linked-files": "^1.5.2",
    "gatsby-remark-images": "^1.5.4",
    "gatsby-remark-prismjs": "^1.2.1",
    "gatsby-remark-responsive-iframe": "^1.4.3",
    "gatsby-source-filesystem": "^1.5.1",
    "gatsby-transformer-remark": "^1.7.15",
    "lodash": "^4.15.0",
    "prop-types": "^15.5.10",
    "remark": "^8.0.0",
    "remark-react": "^4.0.1"
  },
  "devDependencies": {
    "gh-pages": "^0.12.0",
    "prettier": "^1.6.1"
  },
  "homepage": "https://github.com/gatsbyjs/gatsby-starter-blog#readme",
  "keywords": [
    "gatsby"
  ],
  "license": "MIT",
  "main": "n/a",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/gatsbyjs/gatsby-starter-blog.git"
  },
  "scripts": {
    "dev": "gatsby develop",
    "lint": "./node_modules/.bin/eslint --ext .js,.jsx --ignore-pattern public .",
    "test": "echo \"Error: no test specified\" && exit 1",
    "format": "prettier --trailing-comma es5 --no-semi --single-quote --write 'src/**/*.js'",
    "develop": "gatsby develop",
    "build": "gatsby build",
    "deploy": "gatsby build --prefix-paths && gh-pages -d public",
    "fix-semi": "eslint --quiet --ignore-pattern node_modules --ignore-pattern public --parser babel-eslint --no-eslintrc --rule '{\"semi\": [2, \"never\"], \"no-extra-semi\": [2]}' --fix gatsby-node.js"
  }
}
```

`gatsby-node.js`
```javascript
const _ = require('lodash');
const Promise = require('bluebird');
const path = require('path');
const select = require(`unist-util-select`);
const fs = require(`fs-extra`);

exports.createPages = ({graphql, boundActionCreators}) => {
  const {createPage} = boundActionCreators;

  return new Promise((resolve, reject) => {
    const pages = [];
    const blogPost = path.resolve('./src/templates/blog-post.js');
    resolve(
      graphql(
        `
      {
        allMarkdownRemark(limit: 1000) {
          edges {
            node {
              frontmatter {
                path
              }
            }
          }
        }
      }
    `
      ).then(result => {
        if (result.errors) {
          console.log(result.errors);
          reject(result.errors);
        }

        // Create blog posts pages.
        _.each(result.data.allMarkdownRemark.edges, edge => {
          createPage({
            path: 'blog' + edge.node.frontmatter.path,
            component: blogPost,
            context: {
              path: edge.node.frontmatter.path,
            },
          });
        });
      })
    );
  });
};
```

**Comment by arlair** [comment]
Similar issue [here](https://stackoverflow.com/questions/46696753/gatsby-styles-are-not-working-in-production-build/).

Perhaps check if you have an `extract-text-webpack-plugin` compatible with Webpack 1 in `node_modules`.

**Comment by realseanp** [comment]
I've spent DAYS trying to debug this issue and I've finally figured it out. The same instance of the `extract-text-webpack-plugin` must be used in the loader AND the plugin. Gatsby sets up the plugin for you. If you require another `extract-text-webpack-plugin` it might not work. Here is my `gatsby-node.js` file for sass. Notice the `build-css` step where I am splicing out the current ExtractTextPlugin and replacing it with the new instance that's used for the loaders. 

cc @KyleAMathews 

```js
const ExtractTextPlugin = require('extract-text-webpack-plugin');
const { cssModulesConfig } = require('gatsby-1-config-css-modules');
const path = require('path');

exports.modifyWebpackConfig = ({ config, stage }) => {
  const sassFiles = /\.s[ac]ss$/;
  const sassModulesFiles = /\.module\.s[ac]ss$/;
  const sassLoader = `sass?${JSON.stringify({
    includePaths: path.resolve(path.resolve(), '../open/src/'),
  })}`;

  const extractText = new ExtractTextPlugin('styles.css', { allChunks: true });

  switch (stage) {
    case 'develop': {
      config.loader('sass', {
        test: sassFiles,
        exclude: sassModulesFiles,
        loaders: ['style', 'css', sassLoader],
      });

      config.loader('sassModules', {
        test: sassModulesFiles,
        loaders: ['style', cssModulesConfig(stage), sassLoader],
      });

      return config;
    }
    case 'build-css': {
      config.loader('sass', {
        test: sassFiles,
        exclude: sassModulesFiles,
        loader: ExtractTextPlugin.extract(['css?minimize', sassLoader]),
      });

      config.loader('sassModules', {
        test: sassModulesFiles,
        loader: ExtractTextPlugin.extract('style', [
          cssModulesConfig(stage),
          sassLoader,
        ]),
      });

      config._config.plugins.splice(1, 1, extractText);

      return config;
    }
    case 'develop-html':
    case 'build-html':
    case 'build-javascript': {
      config.loader('sass', {
        test: sassFiles,
        exclude: sassModulesFiles,
        loader: 'null',
      });

      config.loader('sassModules', {
        test: sassModulesFiles,
        loader: ExtractTextPlugin.extract('style', [
          cssModulesConfig(stage),
          sassLoader,
        ]),
      });

      return config;
    }
    default: {
      return config;
    }
  }
};
```

**Comment by KyleAMathews** [comment]
Due to the high volume of issues, we're closing out older ones without recent activity. Please open a new issue if you need help!
